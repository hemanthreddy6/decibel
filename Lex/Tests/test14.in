function scaleWave <- (factor: float, t: float): float {
    return SIN(t) * factor;
}

function halfScale <- scaleWave(0.5, _);
audio softWave <- audio(halfScale, 440hz, 5s);

play softWave;
save softWave -> "soft_wave.wav";
