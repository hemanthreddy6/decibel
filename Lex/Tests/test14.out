FUNCTION
IDENTIFIER
<-
(
)
{
IDENTIFIER
<-
INT_LITERAL
;
IF
(
IDENTIFIER
>
INT_LITERAL
)
{
LOOP
IDENTIFIER
OVER
INT_LITERAL
{
IDENTIFIER
<-
IDENTIFIER
-
INT_LITERAL
;
}
}
RETURN
IDENTIFIER
;
}
